src/main/java/todoify/taskmanager/TaskManager.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    /**
+     * Constructor for a task manager, managing a list of items representing "tasks", with the default storage location
+     * and default storage handler.
+     */
+    public TaskManager() {
+        this(InternalPath.of(DEFAULT_FILENAME));
+    }
+
+    /**
+     * Obtains the currently stored tasks as an immutable list.
+     *
+     * @return The list of tasks.
+     */
+    public List<Task> getTasks() {
+        return Collections.unmodifiableList(this.taskList);
+    }
+
+    /**
+     * Queries the number of tasks stored.
+     *
+     * @return The number of tasks currently stored as an integer.
+     */
+    public int getTaskCount() {
+        return this.taskList.size();
+    }
+
+    /**
+     * Adds a task to the task list.
+     *
+     * @param task The task to add.
+     * @return `true`, by {@link List#add} definition.
+     */
+    public boolean addTask(Task task) {
+        return this.taskList.add(task);
+    }
+
+    /**
+     * Gets a task from the task list with the corresponding index.
+     *
+     * @param index The index to obtain.
+     * @return The task in question at the specified index.
+     * @throws IndexOutOfBoundsException if the index is out of bounds.
+     */
+    public Task getTask(int index) {
+        return this.taskList.get(index);
+    }
+
+    /**
+     * Removes the given task from the task list.
+     *
+     * @param task The task to remove.
+     * @return `true` if the task exists in the list and is successfully removed, `false` otherwise.
+     */
+    public boolean removeTask(Task task) {
+        return this.taskList.remove(task);
+    }
+
+    /**
+     * Removes the task at the corresponding index.
+     *
+     * @param index The index to remove.
+     * @return The task in question that was removed.
+     * @throws IndexOutOfBoundsException if the index is out of bounds.
+     */
+    public Task removeTask(int index) {
+        return this.taskList.remove(index);
+    }
+
+    /**
+     * Loads and replaces the task list in memory with the one currently in storage.
+     *
+     * <p>
+     * This method will load the data from storage and replace all in-memory contents. Any unrecognized, incompatible
+     * tasks may be omitted entirely.
+     * </p>
+     *
+     * @throws IOException if there were any issues retrieving the data.
+     */
+    public void loadFromStorage() throws IOException, JsonSyntaxException {
+        Gson gson = new Gson();
+        try {
+            String data = this.storageHandler.loadFrom(this.storageLocation);
+            JsonArray array = JsonParser.parseString(data).getAsJsonArray();
+
+            // Prepare a new list of tasks.
+            List<Task> tasks = new ArrayList<>();
+
+            // Prepare a new set of classes, from most specific to least specific.
+            // This ordering is required to match the provided JSON to a class that's as specific as possible.
+            @SuppressWarnings("unchecked")
```

Comment:
Why do you use @SuppressWarnings here? Perhaps you could write an explanation in the header comment!

Reply:
Thanks for the note, I forgot to include that in the comment. The reason was arrays cannot be initialised with generics, so in this particular case:
```java
Class<Task>[] availClasses = new Class[] { Event.class, Deadline.class, Todo.class };
```
resulted in a warning due to it being an unchecked cast.